http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -93,18 +100,35 @@ class ProxyService(
     .newBuilder()
     .expireAfterWrite(proxyConfig.tokenCacheExpiryTime.toSeconds, TimeUnit.SECONDS)
     .maximumSize(proxyConfig.tokenCacheMaxSize)
+    .recordStats()
     .build(
-      new CacheLoader[String, Future[(UserInfo, Instant)]] {
```

Comment:
Previously this was caching a `Future` which I don't think is ideal. One problem with this is it would cache _all_ requests, even invalid tokens which would cause the `Future` to fail. This may have been filling up the cache prematurely and causing more requests to `tokeninfo`. Now it should only cache _valid_ token lookups, and fail with `AuthenticationError` if we failed to look up the token.